incubator/hnc/internal/reconcilers/object_test.go

```diff
@@ -529,6 +534,54 @@ var _ = Describe("Secret", func() {
 		// The source object in the child shouldn't be deleted since the type has 'Remove' mode.
 		Eventually(hasObject(ctx, "secrets", barName, "bar-sec")).Should(BeTrue())
 	})
+
+	It("should avoid propagating banned annotations", func() {
+		setParent(ctx, barName, fooName)
+		makeObjectWithAnnotation(ctx, "roles", fooName, "foo-annot-role", map[string]string{
+			"annot-a": "value-a",
+			"annot-b": "value-b",
+		})
+
+		// Ensure the object is propagated with both annotations
+		Eventually(func() error {
+			inst, err := getObject(ctx, "roles", barName, "foo-annot-role")
+			if err != nil {
+				return err
+			}
+			annots := inst.GetAnnotations()
+			if annots["annot-a"] != "value-a" {
+				return fmt.Errorf("annot-a: want 'value-a', got %q", annots["annot-a"])
+			}
+			if annots["annot-b"] != "value-b" {
+				return fmt.Errorf("annot-b: want 'value-b', got %q", annots["annot-b"])
+			}
+			return nil
+		}).Should(Succeed(), "waiting for initial sync of foo-annot-role")
+		deleteObject(ctx, "roles", fooName, "foo-annot-role")
```

Comment:
Hmm, why can we delete object in the integration test?

Also, maybe add validation to confirm both the source and the copy are gone before creating new object with the same name, or consider creating object with a different name.

Reply:
It's only namespaces that we can't delete - we've always been able to delete individual objects.

I modified `deleteObject` to only return when the object is fully gone (the source object, in this case). I don't think we need to wait for the copy to be deleted, because the test on line 577 will fail until it's gone - but it's in an Eventually block so we'll just try again until it's been replaced.

Reply:
sg.